### ISSUE TITLE
Compensation : Manage Comp Employees

### ISSUE THREAD

**Issue by afs1jes** [afs1jes/CHNCT-Migration#17]

**Comment by csoroka** [comment]
Can you hide the name column since it isn't used?
When I compare this template to current prod I noticed that some of the columns are not populating with Years 2024, 2025, 2026 and 2027 - can you get that corrected so I can review completely?

Thank you,
Chrissy

**Comment by AVHuntley** [comment]
Two changes made:

- Added "hide" keyword in E1 so that the name column is hidden.
- Modified the Scenario dimension table by sorting the planning scenarios to have years increase in successively. The different sorting order of the scenario members was causing the formulas on the Maint! sheet to evaluate to "Reserved Scenario" instead of a specific year.

When we add new planning scenarios to the system, we should double check to make sure they're getting sorted correctly, or similar problems on this template may occur again.

**Comment by csoroka** [comment]
Thank you for making the formatting changes we requested.
NEW ISSUE:
Within the Manage Comp employees template there are instructions of how to remove an employee (position) from the comp input and manage comp models.  
It states to first to remove the data located in the blue shaded fields in the input model then go into the manage comp model and then click remove row.  I don't believe the "remove row" option is available in the new prod.  Can you please revise the instructions on top to detail out how to remove a position with the new prod?

Thank you,
Chrissy

**Comment by AVHuntley** [comment]
Good call! I updated the instructions on sheet. Do these seem clear to you?

Basically, to add an employee, use the Insert button, add data to the row, then Submit.
To remove an employee, delete the data from that row, then submit.

![image](https://user-images.githubusercontent.com/88676502/204657769-830e9167-7ac8-4bb1-aa3b-8077d9c18e1f.png)

**Comment by csoroka** [comment]
I followed the instructions to remove a position/employee and it didn't work.  I removed the data for position CS87-01 under department 01-300 (as this is the only position that currently needs to be removed) under the Comp input Model then submitted then did the same under the manage comp model and submitted. When I went back into the Comp input Model it is still displaying.

**Comment by AVHuntley** [comment]
I adjusted the Zoom column flags on both this and Compensation Input. I think that should fix the issue.

Instructions shouldn't need to change. To delete an employee, you'll just need to delete all the data on their row, then submit, in both models. Let me know if this seems to be working for you as well now.

**Comment by csoroka** [comment]
Tested this again w/ 300/CS87-01- although it removed the position from the Manage Comp Employees & the Comp input 2 year template - it did not remove it from the Comp input 5yr template.  The position is displaying on the reports as well.

**Comment by csoroka** [comment]
Was this corrected?

**Comment by AVHuntley** [comment]
This has been partially addressed so far, and I plan to keep going (hopefully finish) today.

Done so far - The 2yr Input, 5Yr Input, and Manage templates all have been adjusted to not display removed employees. I've been testing on [CS87-01 Community Health Worker I - InCK Program], so that employee shouldn't show up on these three models.

 In progress - We're making some additional changes to the Input templates to avoid this issue in the future and make it easier to remove employees. The 2yr model has been modified already, now I'm looking at the 5yr model.

**Comment by AVHuntley** [comment]
Question - Is this model used for anything _other_ than 'deleting' employees from the comp input models? With the recent changes to the input models, it may be possible to simply clear employees there instead.

**Comment by csoroka** [comment]
I would use it to delete employees because I know it would remove the employee from both input models.  If I can utilize the comp input model to delete that would be great - if I delete via one model would it delete from both?  Can you give me the steps to delete from the comp input model?

**Comment by AVHuntley** [comment]
I think it should be possible to delete employees from the comp input models, although I'm still on the lookout for ways in which altering this process may have unintended downstream effects. Let's test this procedure:

1. Build the Comp Input model using appropriate selections.
2. Highlight the row containing the employee to be deleted (you can just click on the row number on the far left of the Excel window).
3. Select **A3** > **View** > **Clear** to clear editable cells in the selected row.
4. Select **A3** > **Basic** > **Submit** to submit the current grid (which will include the now-empty cells that were just cleared).

The 2yr model is a subset of the 5yr one. The 2yr affects just the current (forecast) and next (budget) years. The 5yr affects those _and_ the following 3 years. When built, the models should each display all employees with data present in any of the years that model affects.

For example, imagine an employee has data in all five years. Clearing and submitting their row in the 2yr model will clear data for the forecast and budget years and cause subsequent 2yr builds to exclude that employee. But they'll still show up on the 5yr model, since there would still be data for that employee in years three, four, and five.

On the other hand, clearing and submitting from the 5yr model should essentially always prevent an employee record from appearing in future builds of both the 2yr and 5yr comp input models, since that would clear records for all five years at once.

**Comment by csoroka** [comment]
Thank you for the instructions.
I followed them and the job code I added for this testing (CS62-10 under 02-Programs 501-Cooking Matters) was properly removed from the Comp input template 2 year and 5 year and doesn't not display on the Comp Detail report - however I still see the position listed under the Comp Manage Employee template (although there is no salary, FTE, etc data)
Let me know if a step should be added to remove it completely from there as well.
-Chrissy

**Comment by AVHuntley** [comment]
Based on that tip, I made a small modification to the 5yr comp input template. (Added DoNotRead flags over date columns at far right of grid.) Employees deleted in the input models should no longer show up on the Manage Comp Employees model.
